addons/stock/report/report_deliveryslip.xml

```diff
@@ -119,7 +127,21 @@
                                 <!-- If not printing lots/serial numbers => merge lines with same product -->
                                 <t t-else="" name="aggregated_move_lines">
                                     <t t-set="aggregated_lines" t-value="o.move_line_ids._get_aggregated_product_quantities()"/>
-                                        <t t-call="stock.stock_report_delivery_aggregated_move_lines"/>
+                                    <t t-call="stock.stock_report_delivery_aggregated_move_lines"/>
+                                    <t t-set="empty_lines" t-value="o.move_lines.filtered(lambda x: x.product_uom_qty and x.quantity_done == 0)"/>
```

Comment:
By doing that you lost the order and then empty lines will always at the end, it is ok ?